app/scripts/etl-rest-services/etl-rest-services-settings.service.js

```diff
@@ -1,31 +1,66 @@
 /*jshint -W003, -W026, -W117, -W098 */
-(function() {
+(function () {
   'use strict';
 
   angular
-        .module('app.etlRestServices')
-        .service('EtlRestServicesSettings', EtlRestServicesSettings);
+    .module('app.etlRestServices')
+    .service('EtlRestServicesSettings', EtlRestServicesSettings);
 
-  EtlRestServicesSettings.$inject = [];
+  EtlRestServicesSettings.$inject = ['$cookies'];
 
-  function EtlRestServicesSettings() {
+  function EtlRestServicesSettings($cookies) {
     var serviceDefinition;
     var restUrlBaseList = ['https://etl.ampath.or.ke:8002/etl/', 'https://test1.ampath.or.ke:8002/etl/'];
     var restUrlBase = restUrlBaseList[1];
 
+    initialize();
     serviceDefinition = {
-          getCurrentRestUrlBase: getCurrentRestUrlBase,
-          setCurrentRestUrlBase: setCurrentRestUrlBase,
-          restUrlBase: restUrlBase
-        };
+      reInitialize: initialize,
+      getCurrentRestUrlBase: getCurrentRestUrlBase,
+      setCurrentRestUrlBase: setCurrentRestUrlBase,
+      restUrlBase: restUrlBase,
+      addUrlToList: addUrlToList,
+      getUrlBaseList: getUrlBaseList,
+      hasCoockiePersistedCurrentUrlBase: hasCoockiePersistedCurrentUrlBase
+    };
```

Comment:
I think our style requires this to be at the beginning looking like, ideally starting at line 12
var service = {
     // properties
};
return service;